docs/en/Configuration.md

```diff
@@ -242,6 +242,10 @@ Default: `[]`
 An alternative API to setting the `NODE_PATH` env variable, `modulePaths` is an array of absolute paths to
 additional locations to search when resolving modules. Use the `<rootDir>` string token to include the path to your project's root directory. Example: `["<rootDir>/app/"]`.
 
+**NOTE:** An empty `[]` works just fine with how npm works wherein Jest provides necessary configuration. However, if you want to provide this option and use npm, you need to be aware that all npm@3+ versions store flat dependencies under `<rootDir>/node_modules`, and on encountering the need for different versions of the same package, this flat structure is augmented with a non-flat structure like described [here](https://docs.npmjs.com/how-npm-works/npm3).
+To account for all possible `node_modules` folders, you would need to provide `<rootDir>/**/node_modules` (not just `<rootDir>/node_modules`), or else you would run into version conflicts when running tests, while your application code runs like a charm.
```

Comment:
Wait, this option is not a glob pattern, is it? :O

Reply:
@cpojer it seems to work as a glob pattern though. I haven't looked at the code.

Reply:
Oh.. do you mind checking out the code in Jest or deps that use this to confirm it? It's surprising to me, that's why. It may be in Jest's config normalization but that would be surprising.

cc @thymikee please bring clarity

Reply:
I was briefly checking the resolution code yesterday and it looked like it would treat `/root/**/node_modules` as a directory (where `**` is a dir name, not a glob). But didn't have time to dive deeply.

Reply:
Thanks @cpojer @thymikee. after careful examination, seems that the code doesn't treat it like a glob pattern.

I also came across `moduleDirectories`, whose explanation in the docs lines up with standard npm behavior. Perhaps, `modulePaths` was never the right place for this config anyway.

I will close my PR, and the associated issue. If anyone enters into the same hell with `modulePaths` due to similar ignorance, these issues hopefully guide them out of it.

Reply:
<img width="1595" alt="screen shot 2017-10-05 at 1 21 07 pm" src="https://user-images.githubusercontent.com/542168/31240886-1a2e627e-a9d0-11e7-8ddc-ac106648e3f2.png">

Reply:
Normalization [code](https://github.com/facebook/jest/blob/77744a24816d0978b6c478987426c36d615864bd/packages/jest-config/src/normalize.js#L362-L371) which doesn't treat it like a glob.

Not sure why the behavior that I saw when running with the glob pattern, but `<rootDir>/node_modules` surely breaks it for `modulePaths`.